### ISSUE TITLE
Bug: `foundryup` and repeated instances of tagged nightly's without any accompanying binaries 

### ISSUE THREAD

**Issue by sambacha** [foundry-rs/foundry#5679]
### Component

Other (please describe)

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

N/A

### What command(s) is the bug in?

foundryup

### Operating System

None

### Describe the bug

See https://github.com/foundry-rs/foundry/releases/tag/nightly-2e90e35ee52efeba74c06a1484bdc1fb96d01a0e

```console
curl: (22) The requested URL returned error: 403
foundryup: command failed: curl -sSf https://api.github.com/repos/foundry-rs/foundry/git/refs/tags/nightly
foundryup: installing foundry (version nightly, tag nightly-)
foundryup: downloading latest forge, cast, anvil, and chisel
############################################################################################################################################################################# 100.0%
tar: Error opening archive: Unrecognized archive format
foundryup: command failed: tar -xzC /Users/janitor/.foundry/bin
foundryup: downloading manpages
############################################################################################################################################################################# 100.0%
tar: Error opening archive: Unrecognized archive format
foundryup: installed - forge 0.2.0 (06a17bf 2023-08-19T00:24:01.238378000Z)
foundryup: installed - cast 0.2.0 (06a17bf 2023-08-19T00:24:01.088059000Z)
foundryup: installed - anvil 0.2.0 (06a17bf 2023-08-19T00:24:01.389702000Z)
foundryup: installed - chisel 0.2.0 (06a17bf 2023-08-19T00:24:00.932787000Z)
foundryup: done!
```

Will you finally accept a refactor of foundryup to solve all these issues? 
Publishing a `latest.json` for binaries as well would reduce issues when using the github api and getting rate limited (this is the first issue in the command `foundryup: command failed: curl -sSf https://api.github.com/repos/foundry-rs/foundry/git/refs/tags/nightly`

Yes I am running latest foundry up. Can we add versioning to it finally?

**Comment by mattsse** [comment]
fyi @DaniPopes

**Comment by Evalir** [comment]
closed with #5682, ty for reporting
